rocky/reports/views/base.py

```diff
@@ -120,13 +120,24 @@ def get(self, request: HttpRequest, *args: Any, **kwargs: Any) -> HttpResponse:
         return redirect(reverse("report_history", kwargs=self.get_kwargs()))
 
 
+def get_plugin_ids(report_types: list[type[BaseReport]]):
+    return report_plugins_union(report_types)
+
+
 def hydrate_plugins(report_types: list[type["BaseReport"]], katalogus: KATalogusClientV1) -> dict[str, list[Plugin]]:
-    merged_plugins = report_plugins_union(report_types)
+    plugins: dict[str, list[Plugin]] = {"required": [], "optional": []}
+    merged_plugins = get_plugin_ids(report_types)
+
+    required_plugins_ids = list(merged_plugins["required"])
+    optional_plugins_ids = list(merged_plugins["optional"])
+
+    # avoid empty list getting all plugins from KATalogus
+    if required_plugins_ids:
+        plugins["required"] = sorted(katalogus.get_plugins(ids=required_plugins_ids), key=attrgetter("name"))
+    if optional_plugins_ids:
+        plugins["optional"] = sorted(katalogus.get_plugins(ids=optional_plugins_ids), key=attrgetter("name"))
```

Comment:
These are possibly two calls to the list endpoint. You might be better off retrieving all of them in one go and then separate them? E.g. in pseudocode:
```
plugins = get_plugins(ids=required+optional)
required_plugins = filter(lambda x: x in required, plugins)
optional_plugins = filter(lambda x: x in optional, plugins)
...
```

Reply:
Currently, there is no filtered endpoint, and as such this is just a locally cached 'full copy' which we iteratedover twice I'd guess?

Reply:
True! Forgot about the calls indeed